### PR TITLE
log examples and attention weights to file

### PR DESCRIPTION
**Pull request by Ranapop**

### CODE REVIEW

File: examples/cfq/models.py [hunk 1]
```diff
@@ -263,12 +263,19 @@ def decode_step_fn(carry, x):
...
       decode_step_fn(init_carry, inputs[:, 0])
 
-    _, (logits, predictions) = jax_utils.scan_in_dim(
+    _, (logits, predictions, scores) = jax_utils.scan_in_dim(
         decode_step_fn,
         init=init_carry,  # rng, lstm_state, last_pred
         xs=inputs,
         axis=1)
-    return logits, predictions
+    if not self.is_initializing() and not train:
```
Comment (marcvanzee) [inline examples/cfq/models.py:273]: Why did you add this `if` statement? Why not just always return the attention weights?

File: examples/cfq/train.py [hunk 2]
```diff
@@ -338,6 +353,11 @@ def train_model(learning_rate: float = None,
     Do the training on data_source.train_dataset and evaluate on
     data_source.dev_dataset at each epoch and log the results
     """
+  if os.path.isdir(model_dir):
+    shutil.rmtree(model_dir)
```
Comment (marcvanzee) [inline examples/cfq/train.py]: Could you add a comment what this is and why you do it?

File: examples/cfq/train.py [hunk 3]
```diff
@@ -338,6 +353,11 @@ def train_model(learning_rate: float = None,
...
+  os.makedirs(model_dir)
+  logging_file_name = os.path.join(model_dir, 'logged_examples.txt')
+  logging_file = open(logging_file_name,'w')
```
Comment (marcvanzee) [inline examples/cfq/train.py]: This means the file will be opened during the entire time of training, which is not so nice and could go wrong if your training crashes. I would pass `model_dir` to `evaluate_model` below and do the file operations there.

**Review by marcvanzee — CHANGES_REQUESTED** [review]

File: examples/cfq/train.py (hunk 2 above)
Comment (Ranapop) [reply examples/cfq/train.py]: yes

File: examples/cfq/train.py (hunk 3 above)
Comment (Ranapop) [reply examples/cfq/train.py]: won't that impact the speed? opening and closing the file each time I am evaluating the model?

File: examples/cfq/models.py (hunk 1 above)
Comment (Ranapop) [reply examples/cfq/models.py:273]: I am only investigating them during evaluation, so I am avoiding unnecessary operations.

Comment (marcvanzee) [reply examples/cfq/models.py:273]: Ok, could you add a comment explaining that this is the reason?

File: examples/cfq/train.py (hunk 3 above)
Comment (marcvanzee) [reply examples/cfq/train.py]: No that is superfast, and keeping a file open like this is not good practice.

**Review by marcvanzee — APPROVED** [review]
